### DISCUSSION TITLE
Postgres XML type

### DISCUSSION THREAD

**Discussion by joebonneau** [sqlalchemy/sqlalchemy#12135] [Usage Questions]
Hey there,

I was curious what the recommended way would be to implement an XML column for Postgres. It doesn't appear to be a built-in type (unless I've missed something) but has been present in Postgres for many versions.

Thanks in advance!

**Comment by zzzeek** [comment]
for simple round trips [UserDefinedType](https://docs.sqlalchemy.org/en/20/core/custom_types.html#sqlalchemy.types.UserDefinedType) will get you there.   We would assume the PostgreSQL drivers receive and return XML as strings, substitute whatever string/XML routines you want to use 


```py

from YOUR_XML_LIBRARY_HERE import xml_to_string, whatever_postgresql_gives_you_to_xml

import sqlalchemy.types as types

class MyType(types.UserDefinedType):
    cache_ok = True

    def get_col_spec(self, **kw):
        return "XML"

    def bind_processor(self, dialect):
        def process(value):
            value = xml_to_string(value)
            return value
        return process

    def result_processor(self, dialect, coltype):
        def process(value):
            value = whatever_postgresql_gives_you_to_xml(value)
            return value
        return process
```
